File: 00. Estructura/01. Lenguajes de proposito general/02. Programación básica/10. Excepciones/04. Excepciones predefinidas/Java/Excepciones predefinidas.java
package prueba;

public class Prueba {
    
    public static void main(String[] args) {
        
        // Cualquier clase de excepción
        Throwable exception;

        // Excepciones fatales (Hereda de Throwable)
        Error exception;

        // Excepciones no fatales (Hereda de Throwable)
        Exception exception;

        // Excepciones sin marcar (Hereda de Exception)
        RuntimeException exception;
        
        // Error aritmético (Hereda de RuntimeException)
        ArithmeticException exception;

        // Indice de algun tipo fuera de rango (Hereda de RuntimeException)
        IndexOutOfBoundsException exception;

        // Argumento ilegal o inapropiado (Hereda de RuntimeException)
        IllegalArgumentException 

        // Variable nula donde se requiere un objeto (Hereda de RuntimeException)
        NullPointerException exception;

        // Error al convertir cadena a numero (Hereda de IllegalArgumentException)
        NumberFormatException exception;
        
    }
    
}

File: 01. Lenguajes de proposito general/01. Programación básica/11. Excepciones/04. Excepciones predefinidas/Python/Excepciones predefinidas.py

# Cualquier clase de excepción
exception = BaseException
	# Excepciones que no salen del sistema 
	exception = Exception
		# Fallo en la referencia un atributo
		exception = AttributeError
		# Error aritmético
		exception = ArithmeticError
			# El resultado de una operación aritmética es demasiado grande para ser representado
			exception = OverflowError
			# División por 0
			exception = ZeroDivisionError
		# la función input()  alcanza una condición de fin de archivo (EOF) sin leer ningún dato
		exception = EOFError
		# No se encuentra un nombre local o global.
		exception = NameError
		# Una clave o índice utilizado en una asignación o secuencia no es válida
		exception = LookupError
			# Un subíndice de secuencia está fuera de rango
			exception = IndexError
			# No se encuentra una clave en el diccionario
			exception = KeyError
		# El iterador no produce más elementos
		exception = StopIteration
		# Errores realacionados con el sistema operativo
		exception = OSError
		# Cuando una operación o función se aplica a un objeto de tipo inapropiado
		exception = TypeError
		# Cuando una operación o función recibe un argumento que tiene el tipo correcto pero un valor inapropiado
		exception = ValueError
	# Interrupción con el teclado de la ejecucución (ctl+c o delete) 
	exception = KeyboardInterrupt


		
